### PR TITLE
Deprecate --salt-overlayable-shadow-scroll-color

### PR DESCRIPTION
**Pull request by libertymayc**
`--salt-overlayable-shadow-scroll-color` is random to expose, it doesn't align with Figma and shouldn't have been added, better to use shadow color directly

### CODE REVIEW

File: .changeset/funny-kangaroos-relate.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@salt-ds/data-grid": patch
+"@salt-ds/theme": patch
+---
+
+Remove `--salt-overlayable-shadow-scroll-color`
```
Comment (libertymayc) [inline .changeset/funny-kangaroos-relate.md:6]:
```suggestion
Removed `--salt-overlayable-shadow-scroll-color`
```

**Review by JoanaMMoreira — DISMISSED** [review]

**Review by joshwooding — CHANGES_REQUESTED** [review]
This is a breaking change and cannot be made. We should mark this as deprecated in some way and remove it in v2.

File: packages/theme/css/characteristics/overlayable.css [hunk 2]
```diff
@@ -9,4 +8,7 @@
   --salt-overlayable-shadow-modal: var(--salt-shadow-5);
 
   --salt-overlayable-background: var(--salt-palette-neutral-backdrop);
+
+  /* **deprecated**: use --salt-shadow-1-color instead */
+  --salt-overlayable-shadow-scroll-color: var(--salt-shadow-1-color);
```
Comment (Fercas123) [inline packages/theme/css/characteristics/overlayable.css:13]: should we add a deprecation note in the overlayable base docs?

Comment (libertymayc) [reply packages/theme/css/characteristics/overlayable.css:13]: they weren't documented there anyway, so i think unnecessary

**Review by Fercas123 — COMMENTED** [review]
LGTM, just need to run prettier

**Review by Fercas123 — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**